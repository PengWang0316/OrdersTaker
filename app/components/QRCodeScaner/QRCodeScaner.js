import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { PhotoCamera as PhotoCameraIcon } from '@material-ui/icons';
import { Button, Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

import QRScanerDialog from './QRScanerDialog';

const styles = {
  tableNumber: {
    fontSize: 35,
    fontWeight: '800'
  }
};

/**
 * The component that will be used to scan the QR code.
 */
export class QRCodeScaner extends Component {
  static propTypes = {
    orders: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired
  };

  state = {
    isDialogOpen: false
  };

  /**
   * Toggle the dialog open state
   * @return {null} No return
   */
  toggleDialog = () => this.setState(({ isDialogOpen }) => ({ isDialogOpen: !isDialogOpen }));

  /**
 * The render method
 * @return {jsx} Return jsx for the component.
 */
  render() {
    const { orders, classes } = this.props;
    const { isDialogOpen } = this.state;
    return (
      <div>
        {!orders.tableNumber && (
          <Fragment>
            <Button onClick={this.toggleDialog}><PhotoCameraIcon color="primary" /></Button>
            <Typography color="primary">Scan Table QR</Typography>
          </Fragment>
        )}
        {orders.tableNumber && (
          <Fragment>
            <Typography color="primary">Table number:</Typography>
            <Button color="primary" className={classes.tableNumber} onClick={this.toggleDialog}>{orders.tableNumber}</Button>
          </Fragment>
        )}
        <QRScanerDialog isOpen={isDialogOpen} onClose={this.toggleDialog} />
      </div>
    );
  }
}
const mapStateToProps = state => ({
  orders: state.orders
});
export default connect(mapStateToProps, null)(withStyles(styles)(QRCodeScaner));
