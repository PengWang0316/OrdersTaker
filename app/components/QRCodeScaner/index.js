import QRCodeScaner from './QRCodeScaner';

export default QRCodeScaner;
