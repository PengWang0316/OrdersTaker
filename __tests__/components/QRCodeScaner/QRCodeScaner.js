import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { QRCodeScaner } from '../../../app/components/QRCodeScaner/QRCodeScaner';

jest.mock('@material-ui/core/Button', () => 'Button');
jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('@material-ui/icons/PhotoCamera', () => 'PhotoCamera');
jest.mock('../../../app/components/QRCodeScaner/QRScanerDialog', () => 'QRScanerDialog');

describe('QRCodeScaner', () => {
  const defaultProps = {
    classes: {
      tableNumber: 'tableNumber'
    },
    orders: {}
  };
  const getShallowComponent = (props = defaultProps) => shallow(<QRCodeScaner {...props} />);

  test('Initial state and toggleDialog function', () => {
    const component = getShallowComponent();
    expect(component.state('isDialogOpen')).toBe(false);
    component.instance().toggleDialog();
    expect(component.state('isDialogOpen')).toBe(true);
  });

  test('Snapshot without tableNumber', () => expect(renderer.create(<QRCodeScaner {...defaultProps} />).toJSON()).toMatchSnapshot());
  test('Snapshot with tableNumber', () => expect(renderer.create(<QRCodeScaner {...{ ...defaultProps, orders: { tableNumber: 1 } }} />).toJSON()).toMatchSnapshot());
});
