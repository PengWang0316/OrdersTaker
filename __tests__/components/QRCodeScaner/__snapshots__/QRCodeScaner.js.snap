// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCodeScaner Snapshot with tableNumber 1`] = `
<div>
  <Typography
    color="primary"
  >
    Table number:
  </Typography>
  <Button
    className="tableNumber"
    color="primary"
    onClick={[Function]}
  >
    1
  </Button>
  <QRScanerDialog
    isOpen={false}
    onClose={[Function]}
  />
</div>
`;

exports[`QRCodeScaner Snapshot without tableNumber 1`] = `
<div>
  <Button
    onClick={[Function]}
  >
    <PhotoCamera
      color="primary"
    />
  </Button>
  <Typography
    color="primary"
  >
    Scan Table QR
  </Typography>
  <QRScanerDialog
    isOpen={false}
    onClose={[Function]}
  />
</div>
`;
